pytorch/src/main/resources/org/bytedeco/pytorch/include/platform_unification.h

```diff
@@ -0,0 +1,21 @@
+/* Add the necessary C++ calls to present a unified C++ API whatever the platform */
+
+namespace javacpp {
+
+   inline c10::Half *allocate_Half(jfloat value) {
+#if defined(__aarch64__) && !defined(C10_MOBILE) && !defined(__CUDACC__)
+       return new c10::Half((float16_t) value);
```

Comment:
Please don`t create a header file just for this. Just use `@Cast()` from JavaCPP

Reply:
For the constructor, we need to cast the float coming from Java to a float16_t on ARM64 only. 
So something like this could do the trick, as a replacement for the header file:
```Java
  private native void allocate(@Cast("/**/\n#if defined(__aarch64__) && !defined(C10_MOBILE) && !defined(__CUDACC__)\nfloat16_t\n#else\nfloat\n#endif\n/**/") float value);
```
Or do you have something else in mind ?

Reply:
Yeah, just a normal `@Cast("float16_t") float`. What kind of error do you get with that??

Reply:
```
error: 'float16_t' was not declared in this scope
```
on non-arm64 platforms. Why would you expect anything else ?

Reply:
Of course, on other platforms, don't use the cast

Reply:
You mean generating different Java code for each platform ? A code that will end up in the shared Java jar ?
I guess that it's not a big deal as long the change only concerns JavaCPP annotations and not the Java API itself,
and that all github workers run the parser instead of using what's in `gen` directory of the repo.
But is it really better that the `#ifdef` solution ?

Reply:
Yes, something like that. I think it's better because we can write it in Java.

Reply:
I agree than the `ifdef` in the info is quite ugly. That's why I thought about a specific header file to bridge between a unique Java API and changes in C++ API.
In such header, we can use the exact same `#ifdef` clauses that the one used in pytorch source and it's quite readable.
There are other things we could add there. For instance I skipped `getCurrentCUDABlasLtHandle` for all platforms because Pytorch does not include the function when compiling with VS for some reason. Probably nobody will ever try to call this from Java, but if it's needed, we could add a wrapper with proper `#ifdef` and either return null or throws an exception in the case we compiled with VS. We cannot do this in Java because we cannot declare a native function only for some platforms and keep the same java bytecode.

Reply:
APIs that differ like that between platform that isn't portable in the first place probably isn't part of the actual intended external API, so let's just skip those.